controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java

```diff
@@ -818,6 +818,115 @@ public void testUserInstancesNotInDeploymentSpec() {
+        assertEquals(Optional.empty(), i3.instance().change().application());
+        assertEquals(Optional.empty(), i4.instance().change().application());
+
+        // Tests run in i4, as they're declared there, and i1 and i2 get to work
+        i4.runJob(systemTest).runJob(stagingTest);
+        i1.runJob(productionUsEast3);
+        i2.runJob(productionUsEast3);
+
+        // Since the post-deployment delay of i1 is incomplete, i3 doesn't yet get the change.
+        tester.outstandingChangeDeployer().run();
+        assertEquals(v0, i1.instance().latestDeployed());
+        assertEquals(v0, i2.instance().latestDeployed());
+        assertEquals(Optional.empty(), i1.instance().change().application());
+        assertEquals(Optional.empty(), i2.instance().change().application());
+        assertEquals(Optional.empty(), i3.instance().change().application());
+        assertEquals(Optional.empty(), i4.instance().change().application());
+
+        // When the delay is done, i3 gets the change.
+        tester.clock().advance(Duration.ofHours(6));
+        tester.outstandingChangeDeployer().run();
+        assertEquals(Optional.empty(), i1.instance().change().application());
+        assertEquals(Optional.empty(), i2.instance().change().application());
+        assertEquals(v0, i3.instance().change().application());
+        assertEquals(Optional.empty(), i4.instance().change().application());
+
+        // v0 begins roll-out in i3, and v1 is submitted and rolls out in i1 and i2 some time later
+        i3.runJob(productionUsEast3); // v0
+        tester.clock().advance(Duration.ofHours(12));
+        i1.submit(applicationPackage);
+        Optional<ApplicationVersion> v1 = i1.lastSubmission();
+        i4.runJob(systemTest).runJob(stagingTest);
+        i1.runJob(productionUsEast3); // v1
+        i2.runJob(productionUsEast3); // v1
+        assertEquals(v1, i1.instance().latestDeployed());
+        assertEquals(v1, i2.instance().latestDeployed());
+        assertEquals(Optional.empty(), i1.instance().change().application());
+        assertEquals(Optional.empty(), i2.instance().change().application());
+        assertEquals(v0, i3.instance().change().application());
+        assertEquals(Optional.empty(), i4.instance().change().application());
+
+        // After some time, v2 also starts rolling out to i1 and i2, but does not complete in i2
+        tester.clock().advance(Duration.ofHours(3));
+        i1.submit(applicationPackage);
+        Optional<ApplicationVersion> v2 = i1.lastSubmission();
+        i4.runJob(systemTest).runJob(stagingTest);
+        i1.runJob(productionUsEast3); // v2
+        tester.clock().advance(Duration.ofHours(3));
+
+        // v1 is all done in i1 and i2, but does not yet roll out in i3; v2 is not completely rolled out there yet.
+        // TODO jonmv: thie belowh new revision policy, but must be faked for now, as v1 would not wait for v0 to complete.
```

Comment:
"thie belowh"? 😄

Reply:
Horrible keyboardss on the 2018 models :)